shell: openthread/shell/boards folder creation to support MCXW7x

xavraz opened the pull request:
Creation of the zephyr/samples/net/openthread/shell/boards to support the MCXW7x boards components.

andrei-menzopol commented on dts/arm/nxp/nxp_mcxw71.dtsi:
```diff
@@ -24,17 +24,11 @@
-	stcm1: system_memory@1a000 {
-		compatible = "zephyr,memory-region", "mmio-sram";
-		reg = <0x1a000 DT_SIZE_K(8)>;
-		zephyr,memory-region = "RetainedMem";
```
These 8KB is from independent power switch domain which can keep data retention in all power modes. So it should be left for low-power use cases, if any. I suggest creating a board overlay.
I faced the same issue here: https://github.com/zephyrproject-rtos/zephyr/pull/83195#discussion_r1941153619

rlubos reviewed (commented):
Please prefix the commit title with `samples: openthread: shell:`, the current prefix suggests the change is related to the shell subsystem which isn't the case. 
Also the DTS change seems unrelated to the sample changes, should go in a separate commit.

xavraz committed:
samples: openthread: shell: boards folder creation to support MCXW7x

Creation of the zephyr/samples/net/openthread/shell/boards to support
the MCXW7x boards components.

Signed-off-by: Xavier Razavet <xavier.razavet@nxp.com>

xavraz committed:
dts: arm: nxp: MCXW71 stcm RAM size increased to 112K

To prevent a RAM overflow in the zephyr/samples/net/openthread/shell
application

Signed-off-by: Xavier Razavet <xavier.razavet@nxp.com>

xavraz commented:
> Please prefix the commit title with `samples: openthread: shell:`, the current prefix suggests the change is related to the shell subsystem which isn't the case. Also the DTS change seems unrelated to the sample changes, should go in a separate commit.

Done with another commit related to the RAM purpose.

xavraz commented on samples/net/openthread/shell/README.rst:
```diff
@@ -46,9 +46,21 @@ Example building for NXP's RW612 FRDM (RCP host).
+   :conf: "prj-ot-host-nxp.conf"
    :goals: build
    :compact:
+west build -b frdm_rw612 zephyr/samples/net/openthread/shell -d <build_path> -p -- -DEXTRA_CONF_FILE="prj-ot-host-nxp.conf"
```
removed

rlubos reviewed (dismissed):
There are some minor compliance issues so please check the report.

axelnxp commented on samples/net/openthread/shell/boards/frdm_rw612.conf:
```diff
@@ -0,0 +1,15 @@
+
+# IMU & FW loader
+CONFIG_NXP_RF_IMU=y
+CONFIG_NXP_FW_LOADER=y
```
same remark here

axelnxp reviewed (changes requested):
I think the FW_LOADER and RF_IMU kconfig should be select by 15.4 and not at application level, other than that looks good to me.

axelnxp reviewed (approved):
approving since my comments are just a suggestion and not blocker
